### ISSUE TITLE
bug: DialogHTMLAttributes does not contain onCancel event

### ISSUE THREAD

**Issue by FabioGimmillaro** [ionic-team/stencil#4267]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.22.3

### Current Behavior

The DialogHTMLAttribute definition does not contain the onCancel event of the dialog element.

### Expected Behavior

The onCancel event is defined and can be used

### System Info

```shell
Irrelevant
```

### Steps to Reproduce

Create a new Stencil-Project
Add a new component.

Add following code to the render-function:
```tsx
return <dialog onCancel={(e) => console.log(e)}></dialog>
```

=> "Type '{ onCancel: (e: any) => void; }' is not assignable to type 'DialogHTMLAttributes<HTMLDialogElement>'"

### Code Reproduction URL

https://github.com/bassettsj/stencil-dialog-element-bug

### Additional Information

I checked for other issues and there is one which was resolved 3 years ago: https://github.com/ionic-team/stencil/issues/1049
The "Code Reproduction URL" within my issue is copied from the old issue.

Unfortunately this doesn't work for our current StencilJS version.
I also checked on your main branch which also doesn't define onCancel within the stencil-public-runtime file: https://github.com/ionic-team/stencil/blob/6c5c98456bee5e1eb706af9b15be3f4f6235144a/src/declarations/stencil-public-runtime.ts#L907

**Comment by tanner-reits** [comment]
Hey @FabioGimmillaro 👋 

Thanks for reporting this issue! I was able to confirm that the bindings for `onClose` and `onCancel` are not available on the `dialog` element. I'll get this labeled to be added to our backlog!

**Comment by rwaskiewicz** [comment]
👋 

The fix for this issue has been included in the [🏒 v3.2.2 release of Stencil](https://github.com/ionic-team/stencil/releases/tag/v3.2.2), which was released earlier today. As a result, I'm going to close this issue. Should it reappear, please feel free to open a new issue. Thanks!
